vizro-core/tests/unit/vizro/figures/test_kpi_cards.py

```diff
@@ -0,0 +1,104 @@
+    def test_kpi_card_mandatory(self):
+        result = kpi_card(data_frame=df, value_column="Actual")()
+        assert_component_equal(
+            result,
+            dbc.Card(
+                [dbc.CardHeader([None, html.H2("Sum Actual")]), dbc.CardBody("73902")],
+                className="card-kpi",
+            ),
+        )
+
+    def test_kpi_card_mandatory_and_optional(self):
+        result = kpi_card(
+            data_frame=df,
+            value_column="Actual",
+            icon="shopping_cart",
+            title="sales",
+            value_format="${value:0.2f}",
+            agg_func="mean",
+        )()
+        assert_component_equal(
+            result,
+            dbc.Card(
+                [
+                    dbc.CardHeader([html.P("shopping_cart", className="material-symbols-outlined"), html.H2("sales")]),
+                    dbc.CardBody("$24634.00"),
+                ],
+                className="card-kpi",
+            ),
+        )
+
+    def test_kpi_card_reference_mandatory(self):
+        result = kpi_card_reference(data_frame=df, value_column="Actual", reference_column="Reference")()
+        expected = dbc.Card(
+            [
+                dbc.CardHeader(
+                    [
+                        None,
+                        html.H2("Sum Actual"),
+                    ]
+                ),
+                dbc.CardBody("73902"),
+                dbc.CardFooter(
+                    [
+                        html.Span("arrow_circle_up", className="material-symbols-outlined"),
+                        html.Span("2.4% vs. reference (72159)"),
+                    ],
+                    className="color-pos",
+                ),
+            ],
+            className="card-kpi",
+        )
+
+        assert_component_equal(result, expected)
+
+    def test_kpi_card_reference_mandatory_and_optional(self):
+        result = kpi_card_reference(
+            data_frame=df,
+            value_column="Actual",
+            reference_column="Reference",
+            icon="shopping_cart",
+            title="sales",
+            value_format="${value:0.2f}",
+            reference_format="+{delta:.1f} vs. {reference:.1f}",
+            agg_func="mean",
+        )()
+        expected = dbc.Card(
+            [
+                dbc.CardHeader(
+                    [
+                        html.P("shopping_cart", className="material-symbols-outlined"),
+                        html.H2("sales"),
+                    ]
+                ),
+                dbc.CardBody("$24634.00"),
+                dbc.CardFooter(
+                    [
+                        html.Span("arrow_circle_up", className="material-symbols-outlined"),
+                        html.Span("+581.0 vs. 24053.0"),
+                    ],
+                    className="color-pos",
+                ),
+            ],
+            className="card-kpi",
+        )
+
+        assert_component_equal(result, expected)
+
```

Comment:
I think you could use a couple more tests here since the reference card has a few extra conditionals and edge cases to check:
* `delta<0`
* `delta=0` (where we actually never discussed what should happen but currently gets treated as `delta<0`)
* `reference=0`

Reply:
Good point! I've changed the styling for delta=0 - there shouldn't be any colour applied.

![Screenshot 2024-06-21 at 11 01 19](https://github.com/mckinsey/vizro/assets/90609403/02eb7cae-5980-4160-9c11-d46ec6decaa2)